### PR TITLE
feat: add a page with a feature matrix / checklist for SDK completeness

### PR DESCRIPTION
**Pull request by cprice404**
This page can be used by developers to make sure they've completed
all of the important functionality before we move an SDK to 1.0.

### CODE REVIEW

**Comment by cprice404** [comment]
Use the "rich diff" button to see the rendered markdown.

**Comment by cprice404** [comment]
@NoSQLKnowHow @driftdiver you may be interested in this.

File: docs/client-specifications/feature-matrix.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+# Momento Client SDKs: Feature Matrix
+
+This page should serve as a checklist for developers who are preparing the client library
+for a given language for a 1.0 release.
+
+## Features / Requirements
+
+* `Release`: Repo has been set up according to our [SDK release process](https://github.com/momentohq/standards-and-practices/blob/main/docs/momento-sdk-release-process.md) and binary artifacts are published to the appropriate
+             package repository for the language.
+* `Get/Set APIs`: Support for basic get/set APIs, with Strings or byte arrays.
```
Comment (schwern) [inline docs/client-specifications/feature-matrix.md:12]: Note that GetResponse should support both value_string and value_bytes?

Comment (schwern) [inline docs/client-specifications/feature-matrix.md:12]: Also cache create, delete, and list.

Note that AlreadyExists is not an error from CreateCache.

File: docs/client-specifications/feature-matrix.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+* `Examples`: Repo includes example code that is tested in CI
+* `Error Handling`: SDK has been updated to be consistent with [Momento error handling specifications](https://github.com/momentohq/standards-and-practices/blob/main/docs/client-specifications/error-handling.md).
+* `Config`: Provides Configuration interface that can be used to adjust relevant settings as per [Momento configuration specification](https://github.com/momentohq/standards-and-practices/blob/main/docs/client-specifications/configuration.md).
+* `Retries`: Basic retry support for failed requests, and the ability for a user to configure a custom retry strategy.
```
Comment (schwern) [inline docs/client-specifications/feature-matrix.md:16]: Do we have an S&P or Notion doc for this?

File: docs/client-specifications/feature-matrix.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+* `LoadGen`: Includes some very minimal scripts that can be used to generate load through the SDK; these are intended for
+             use in tuning for pre-built configurations, and also to serve as a starting point for users who might want to
+            modify them to simulate their own load.
+* `Pre-Built Configs`: Pre-built configurations for Laptop and In-Region, which have been tuned to ensure good performance in
```
Comment (schwern) [inline docs/client-specifications/feature-matrix.md:20]: S&P or Notion doc for this?

File: docs/client-specifications/feature-matrix.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+## Features / Requirements
+
+* `Release`: Repo has been set up according to our [SDK release process](https://github.com/momentohq/standards-and-practices/blob/main/docs/momento-sdk-release-process.md) and binary artifacts are published to the appropriate
+             package repository for the language.
+* `Get/Set APIs`: Support for basic get/set APIs, with Strings or byte arrays.
+* `Examples`: Repo includes example code that is tested in CI
+* `Error Handling`: SDK has been updated to be consistent with [Momento error handling specifications](https://github.com/momentohq/standards-and-practices/blob/main/docs/client-specifications/error-handling.md).
+* `Config`: Provides Configuration interface that can be used to adjust relevant settings as per [Momento configuration specification](https://github.com/momentohq/standards-and-practices/blob/main/docs/client-specifications/configuration.md).
+* `Retries`: Basic retry support for failed requests, and the ability for a user to configure a custom retry strategy.
+* `LoadGen`: Includes some very minimal scripts that can be used to generate load through the SDK; these are intended for
```
Comment (schwern) [inline docs/client-specifications/feature-matrix.md:17]: S&P or Notion doc?

File: docs/client-specifications/feature-matrix.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+             use in tuning for pre-built configurations, and also to serve as a starting point for users who might want to
+            modify them to simulate their own load.
+* `Pre-Built Configs`: Pre-built configurations for Laptop and In-Region, which have been tuned to ensure good performance in
+               each environment.
+* `Logging`: Supports logging configuration; logging at different verbosity levels and to different destinations, ideally
```
Comment (schwern) [inline docs/client-specifications/feature-matrix.md:22]: S&P or Notion doc?

File: docs/client-specifications/feature-matrix.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+             via the idiomatic logging framework for the language (if one exists).
+* `Debug Logging`: includes log statements for logging the execution and success or failure of every API call, for debugging
```
Comment (schwern) [inline docs/client-specifications/feature-matrix.md:24]: S&P or Notion doc? How is "Logging" distinct from "Debug Logging"?

**Review by schwern — COMMENTED** [review]
Other sections to consider. Maybe they need their own S&P.

* Testing
  * Unit
  * Acceptance (live)
* Documentation
  * README template
  * Method docs and examples

**Review by pgautier404 — APPROVED** [review]
Looks good!

File: docs/client-specifications/feature-matrix.md (hunk 6 above)
Comment (cprice404) [reply docs/client-specifications/feature-matrix.md:24]: I just need names that I can use in the matrix.  Do the descriptions above not make it clear what the difference is?  One is basic logging integration with the language's logging platform; the other is actual trace logs for every api call.

File: docs/client-specifications/feature-matrix.md [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+| LoadGen           | :white_check_mark: | :x:                | :white_check_mark: | :white_check_mark: | :x:                | :x:                | :x:                | :x:                |
+| Pre-built Configs | :white_check_mark: | :x:                | :x:                | :x:                | :x:                | :x:                | :x:                | :x:                |
+| Logging           | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: | :x:                | :x:                | :x:                | :x:                |
+| Debug Logging     | :white_check_mark: | :x:                | :x:                | :x:                | :x:                | :x:                | :x:                | :x:                |
+| Lambda            | :x:                | :x:                | :x:                | :x:                | :x:                | :x:                | :x:                | :x:                |
+| Data Structures   | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: | :x:                | :x:                | :x:                | :x:                |
```
Comment (eaddingtonwhite) [inline docs/client-specifications/feature-matrix.md:48]: As we expand datastructure apis maybe this should be multiple sections or specific datastructure apis? Just could see slightly different completion rates between different APIs. Can wait to address till issue though not blocker for now.

**Review by eaddingtonwhite — APPROVED** [review]
Love this! Super helpful!

File: docs/client-specifications/feature-matrix.md [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+* `Debug Logging`: includes log statements for logging the execution and success or failure of every API call, for debugging
+                   purposes.  Should only be visible at TRACE log level.
+* `Lambda`: SDK has been tested for use in AWS Lambda environment.  This includes the following:
+  * The SDK should provide a pre-built configuration that takes into account that Lambda environments may require a longer
+    timeout to establish their initial connection due to cold start considerations.
+  * Also requires testing to ensure that if the client is stored in a global variable, and is re-used after more than 350s
+    of idle time (the idle connection timeout for AWS NLB), the client will be able to gracefully re-establish the gRPC connection.
+* `Data Structures`: supports APIs for Lists, Sets, and Dictionaries
+
+## Current Status
```
Comment (poppoerika) [inline docs/client-specifications/feature-matrix.md:33]: ❤️

**Review by poppoerika — DISMISSED** [review]

**Comment by cprice404** [comment]
I added a commit https://github.com/momentohq/standards-and-practices/pull/39/commits/686ff11e68033c8587192b09e2d802ecca53cf55 to address most of the feedback from the PR; all of the things that I thought were actionable / good ROI at this time.

**Review by poppoerika — APPROVED** [review]

**Comment by cprice404** [comment]
@schwern I'm gonna go ahead and merge this even though I didn't get a chance to hear your response to my reply about linking to the specs repos for the deep details, since you are out for the next few days.

we can discuss further / iterate on this going forward though.

File: docs/client-specifications/feature-matrix.md (hunk 6 above)
Comment (schwern) [reply docs/client-specifications/feature-matrix.md:31]: Got it.

**Comment by cprice404** [comment]
@schwern I did add integration testing to the list btw.  Thanks for catching that.

**Pull request merged**